libbeat/cmd/instance/metrics.go

```
@@ -36,3 +59,35 @@ func reportInfo(_ monitoring.Mode, V monitoring.Visitor) {
 	uptime := int64(delta.Seconds())*1000 + int64(delta.Nanoseconds()/1000000)
 	monitoring.ReportInt(V, "uptime.ms", uptime)
 }
+
+func reportCPU(_ monitoring.Mode, V monitoring.Visitor) {
+	V.OnRegistryStart()
+	defer V.OnRegistryFinished()
+
+	cpuUsage, normalizedCPU := getCPUUsage()
+	monitoring.ReportFloat(V, "usage", cpuUsage)
+	monitoring.ReportFloat(V, "usage.normalized", normalizedCPU)
+}
+
+func getCPUUsage() (float64, float64) {
+	pid := os.Getpid()
+
+	sample := cpuSample{
+		time:      time.Now(),
+		procTimes: sigar.ProcTime{},
+	}
+
+	if err := sample.procTimes.Get(pid); err != nil {
```

I'd prefer to return the error and have the caller handle it.